File: ChatGPT(예진)/Python (수정 완료)/Number_baseball_Game.py
import random

def generate_random_number():
    return random.sample(range(1, 10), 5)

def check_guess(random_number, user_guess):
    strikes = 0
    balls = 0
    for i in range(len(random_number)):
        if user_guess[i] == random_number[i]:
            strikes += 1
        elif user_guess[i] in random_number:
            balls += 1
    return strikes, balls

def main():
    print("Let's start the Number Baseball Game!")
    print("Game Rules: Guess a 5-digit number with digits from 1 to 9 without repetition.")

    random_number = generate_random_number()
    attempts = 0
    game_record = []

    while True:
        user_input = input("Enter a 5-digit number (type 'q' to quit): ")
        if user_input.lower() == 'q':
            break
        if len(user_input) != 5 or not user_input.isdigit():
            print("Invalid input. Please enter a 5-digit number.")
            continue

        user_guess = [int(digit) for digit in user_input]
        attempts += 1
        strikes, balls = check_guess(random_number, user_guess)

        if strikes == 5:
            print(f"Congratulations! You guessed the correct answer in {attempts} attempts!")
            break

        print(f"{strikes} Strike, {balls} Ball")
        game_record.append((user_input, strikes, balls))

    print("Game Records:")
    print("Guess\t\tStrikes\tBalls")
    for record in game_record:
        print(f"{record[0]}\t\t{record[1]}\t\t{record[2]}")

if __name__ == "__main__":
    main()

File: Copilot(예진)/Java/Number_baseball_Game.java
// 숫자야구 게임을 개발해 주세요.

import java.util.Scanner;

public class Number_baseball_Game {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("숫자야구 게임을 시작합니다.");
        System.out.println("게임을 시작하려면 1을 입력하세요.");
        int start = sc.nextInt();
        if (start == 1) {
            System.out.println("게임을 시작합니다.");
            int[] answer = new int[3];
            for (int i = 0; i < 3; i++) {
                answer[i] = (int) (Math.random() * 9) + 1;
                for (int j = 0; j < i; j++) {
                    if (answer[i] == answer[j]) {
                        i--;
                        break;
                    }
                }
            }

            int[] guess = new int[3];
            int strike = 0;
            int ball = 0;
            int count = 0;
            while (strike != 3) {
                strike = 0;
                ball = 0;
                System.out.println("세 자리 숫자를 입력하세요.");
                int num = sc.nextInt();
                guess[0] = num / 100;
                guess[1] = (num % 100) / 10;
                guess[2] = num % 10;
                for (int i = 0; i < 3; i++) {
                    for (int j = 0; j < 3; j++) {
                        if (answer[i] == guess[j]) {
                            if (i == j) {
                                strike++;
                            } else {
                                ball++;
                            }
                        }
                    }
                }
                System.out.println("Strike: " + strike + ", Ball: " + ball);
                count++;
            }
            System.out.println("정답입니다. " + count + "번 만에 맞추셨습니다.");
        } else {
            System.out.println("게임을 종료합니다.");
        }
    }
}